src/main/java/duckie/Storage.java

```diff
@@ -0,0 +1,150 @@
+package duckie;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duckie.exception.DuckieException;
+import duckie.exception.DuckieFileErrorException;
+import duckie.task.Deadline;
+import duckie.task.Event;
+import duckie.task.Task;
+import duckie.task.Todo;
+
+
+
+/**
+ * Deals with the stored duckie file in the HardDrive
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Instantiate a Storage object
+     * @param filePath Path of duckie.txt
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Opens the duckie file from the harddrive
+     * Creates the file and directory if cannot be found
+     * @return File containing the saved tasks
+     * @throws DuckieException
+     */
+    public File openFile() throws DuckieException {
+        try {
+            File duckieFile = new File(this.filePath);
+            if (!duckieFile.exists()) {
+                File dir = new File(duckieFile.getParent());
+                if (!dir.exists()) {
+                    if (dir.mkdirs() && duckieFile.createNewFile()) {
```

Comment:
One suggestion here could be to consider the fact that the mkdirs and createNewFile functions already make a new directory/file if it doesnt exist.